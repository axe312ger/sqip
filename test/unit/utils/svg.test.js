const SVGO = require('svgo')

const { runSVGO, replaceSVGAttrs } = require('../../../src/utils/svg')

jest.mock('svgo')

test('runSVGO', () => {
  const inputSVG = '<svg />'
  runSVGO(inputSVG)
  expect(SVGO).toHaveBeenCalledTimes(1)
  expect(SVGO).toHaveBeenLastCalledWith({ multipass: true, floatPrecision: 1 })
  expect(SVGO.mock.instances[0].optimize).toHaveBeenCalledTimes(1)
  expect(SVGO.mock.instances[0].optimize.mock.calls[0][0]).toBe(inputSVG)
  // @todo switch to async svgo, this will allow proper testing
})

describe('replaceSVGAttrs', () => {
  test('svg with group, no config', () => {
    const result = replaceSVGAttrs(
      '<svg><path fill="#bada55" d="M0 0h1024v640H0z"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>',
      {}
    )
    expect(result).toMatchSnapshot()
  })
  test('svg with group, config with dimensions only', () => {
    const result = replaceSVGAttrs(
      '<svg><path fill="#bada55" d="M0 0h1024v640H0z"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>',
      { width: 1024, height: 640 }
    )
    expect(result).toMatchSnapshot()
  })
  test('svg with group, config with dimensions only and blur', () => {
    const result = replaceSVGAttrs(
      '<svg><path fill="#bada55" d="M0 0h1024v640H0z"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>',
      { width: 1024, height: 640, blur: 5 }
    )
    expect(result).toMatchSnapshot()
  })
  test('svg with group, config with dimensions and zero blur', () => {
    const result = replaceSVGAttrs(
      '<svg><path fill="#bada55" d="M0 0h1024v640H0z"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>',
      { width: 1024, height: 640, blur: 0 }
    )
    expect(result).toMatchSnapshot()
  })
  test('svg without group, config with dimensions only', () => {
    const result = replaceSVGAttrs(
      '<svg><path fill="#bada55" d="M0 0h1024v640H0z"/><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></svg>',
      { width: 1024, height: 640 }
    )
    expect(result).toMatchSnapshot()
  })
})
