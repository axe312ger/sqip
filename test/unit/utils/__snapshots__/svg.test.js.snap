// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceSVGAttrs svg with group, config with dimensions and zero blur 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 640\\"><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`replaceSVGAttrs svg with group, config with dimensions only 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 640\\"><filter id=\\"b\\"><feGaussianBlur stdDeviation=\\"12\\" /></filter><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g filter=\\"url(#b)\\"><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`replaceSVGAttrs svg with group, config with dimensions only and blur 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 640\\"><filter id=\\"b\\"><feGaussianBlur stdDeviation=\\"5\\" /></filter><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g filter=\\"url(#b)\\"><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`replaceSVGAttrs svg with group, no config 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 undefined undefined\\"><filter id=\\"b\\"><feGaussianBlur stdDeviation=\\"12\\" /></filter><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g filter=\\"url(#b)\\"><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`replaceSVGAttrs svg without group, config with dimensions only 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 640\\"><filter id=\\"c\\"><feGaussianBlur stdDeviation=\\"55\\" /></filter><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g filter='url(#c)' fill-opacity='.5'><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;
